### DISCUSSION TITLE
Store Enum.value instead of Enum.name

### DISCUSSION THREAD

**Discussion by RafRaf** [sqlalchemy/sqlalchemy#11527] [Ideas]
I have some code like below:

```python
from enum import Enum

class Feature(enum.Enum):
    POWER = 'pwr'
    SPEED = 'spd'
    STAMINA = 'stm'

class Player(Base):
    ...
    feature: Mapped[Feature] = mapped_column(
        Enum(Feature, native_enum=False), default=Feature.POWER
    )
```

I was 100% sure that sqlalchemy stores Enum.values but it's not. After the query like:

```python
insert(Player)
.values(feature=Feature.SPEED)
```

I see `SPEED` in the table not `spd`. I've found in docs that this behaviour is correct but isn't it reasonable to keep Enum.name for code and use Enum.value for DB?

PS: Of course I can use `Feature.SPEED.value` all around as a workaround but it's tough

**Comment by CaselIT** [comment]
Hi,

You can use this option to change how the enum works: https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Enum.params.values_callable

> I've found in docs that this behaviour is correct but isn't it reasonable to keep Enum.name for code and use Enum.value for DB?

maybe, but it would be an huge backward incompatible change. Also enum value can be int or other values, not only strings. There is an option for it though

**Reply by RafRaf** [reply]
> but it would be an huge backward incompatible change

It's hard to even imagine 🙂

I tried `enum.StrEnum` because its` __str__` method returns the required value, but it doesn't work either. 😐

Anyway, `values_callable` works as expected.
Thank you @CaselIT 👍

**Reply by Oliversinn** [reply]
Hello, I'm facing the same issue, I tried the `values_callable` but couldn't make it work. Can you share the snippet of code you used to fixed it?

**Reply by trendels** [reply]
This should work:

```python
class Player(Base):
    ...
    feature: Mapped[Feature] = mapped_column(
        Enum(
            Feature, native_enum=False, values_callable=lambda e: [x.value for x in e]
        ),
        default=Feature.POWER,
    )
```

Make sure you don't have a typo in the name of the `values_callable` parameter. I noticed that the `sqlalchemy.Enum` constructor silently ignores unknown arguments.
